cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/WikiContentValidatorAndNormalizer.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.wikipedia;
+
+import co.cask.cdap.api.Resources;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.mapreduce.AbstractMapReduce;
+import co.cask.cdap.api.mapreduce.MapReduceContext;
+import co.cask.cdap.api.workflow.Value;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.annotations.SerializedName;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sweble.wikitext.engine.EngineException;
+import org.sweble.wikitext.engine.PageId;
+import org.sweble.wikitext.engine.PageTitle;
+import org.sweble.wikitext.engine.WtEngineImpl;
+import org.sweble.wikitext.engine.config.WikiConfig;
+import org.sweble.wikitext.engine.nodes.EngProcessedPage;
+import org.sweble.wikitext.engine.utils.DefaultConfigEnWp;
+import org.sweble.wikitext.example.TextConverter;
+import org.sweble.wikitext.parser.parser.LinkTargetException;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * MapReduce program to validate and extract content from raw wikidata blobs.
+ * Normalizes the data by converting the Wikitext to plain text for easier analysis in later stages.
+ */
+public class WikiContentValidatorAndNormalizer extends AbstractMapReduce {
+
+  public static final String NAME = WikiContentValidatorAndNormalizer.class.getSimpleName();
+
+  @Override
+  protected void configure() {
+    setName(NAME);
+    setDescription("A MapReduce program that dumps page titles to a dataset.");
+    setMapperResources(new Resources(512));
+    setInputDataset(WikipediaPipelineApp.RAW_WIKIPEDIA_DATASET);
+    setOutputDataset(WikipediaPipelineApp.NORMALIZED_WIKIPEDIA_DATASET);
+  }
+
+  @Override
+  public void beforeSubmit(MapReduceContext context) throws Exception {
+    Job job = context.getHadoopJob();
+    job.setMapperClass(FilterNormalizerMapper.class);
+    job.setNumReduceTasks(0);
+  }
+
+  @Override
+  public void onFinish(boolean succeeded, MapReduceContext context) throws Exception {
+    WorkflowToken workflowToken = context.getWorkflowToken();
+    if (workflowToken != null) {
+      workflowToken.put("result", Value.of(succeeded));
```

Comment:
The idea was to add the same key to the workflow token from most nodes, then use it in verification. In some cases, I also use the result in a condition predicate.